exasol/driver/websocket.py

```diff
@@ -0,0 +1,556 @@
+    e.g. requesting a .rollback() on a connection that does not support transaction
+    or has transactions turned off.
+    """
+
+
+class Connection(Protocol):
+    """
+    Defines a protocol which is compliant with https://peps.python.org/pep-0249/#connection-objects connection objects.
+    """
+
+    def connect(self):
+        """
+        Connect to the database.
+
+        Attention:
+            Addition not required by pep-249.
+        """
+        ...
+
+    def close(self):
+        """
+        Close the connection now (rather than whenever .__del__() is called).
+
+        The connection will be unusable from this point forward;
+        an Error (or subclass) exception will be raised if any operation is attempted with the connection.
+        The same applies to all cursor objects trying to use the connection.
+        Note that closing a connection without committing the changes first will cause an implicit rollback
+        to be performed.
+        """
+        ...
+
+    def commit(self):
+        """
+        Commit any pending transaction to the database.
+
+        Note:
+            If the database supports an auto-commit feature, this must be initially off.
+            An interface method may be provided to turn it back on.
+            Database modules that do not support transactions should implement this method with void functionality.
+        """
+        ...
+
+    def rollback(self):
+        """
+        This method is optional since not all databases provide transaction support.
+
+        In case a database does provide transactions this method causes the database to roll back to the start
+        of any pending transaction. Closing a connection without committing the changes first
+        will cause an implicit rollback to be performed.
+        """
+        ...
+
+    def cursor(self):
+        """
+        Return a new Cursor Object using the connection.
+
+        If the database does not provide a direct cursor concept, the module will have to emulate cursors
+        using other means to the extent needed by this specification.
+        """
+        ...
+
+
+class Cursor(Protocol):
+    """
+    Defines a protocol which is compliant with https://peps.python.org/pep-0249/#cursor-objects cursor objects.
+    """
+
+    @property
+    def arraysize(self):
+        """
+        This read/write attribute specifies the number of rows to fetch at a time with .fetchmany().
+
+        It defaults to 1 meaning to fetch a single row at a time.
+        Implementations must observe this value with respect to the .fetchmany() method,
+        but are free to interact with the database a single row at a time.
+        It may also be used in the implementation of .executemany().
+        """
+        ...
+
+    @property
+    def description(self):
+        """
+        This read-only attribute is a sequence of 7-item sequences.
+
+        Each of these sequences contains information describing one result column:
```

Comment:
I did hard to understand this sentence. 
After reading the test, I hope I understand better, see suggestion:
```suggestion
        This read-only property is a sequence of sequences.

        Each of the inner sequences describes one column of the result set:
```